components/html-block/html-block.js

```diff
@@ -129,14 +129,18 @@ class HtmlBlock extends LitElement {
 			const fragment = template ? document.importNode(template.content, true) : null;
 			if (fragment) {
 
-				const hasMath = !!fragment.querySelector('math');
+				const hasMathML = !!fragment.querySelector('math');
 
 				const temp = document.createElement('div');
 				temp.appendChild(fragment);
 				const fragmentHTML = temp.innerHTML;
 
-				if (hasMath) {
-					await loadMathJax();
+				const hasLatex = D2L.LP && D2L.LP.Web.UI.Flags.Flag('us125413-mathjax-render-latex', true) && /\$\$|\\\(/.test(fragmentHTML);
```

Comment:
Eventually this will depend on a config variable, at which point I imagine we'll need to implement a provider/requestor pattern here to fetch the right LMS information if it's available.

Reply:
I think we should move this down so that can rely on the `||` short-circuit and avoid the regex test in the case where mathML was found.

Reply:
So I don't think we can. Someone can use the equation editor to create an equation _and_ use inline LaTeX in the same block of HTML. In that case, we still need to load the LaTeX renderer.

Reply:
Oh, seems like we are mixing two things here: has math (mathML or LaTeX), and supports LaTeX.  Can we do something like this:
```javascript
const isLaTeXSupported = D2L.LP && D2L.LP.Web.UI.Flags.Flag('us125413-mathjax-render-latex', true);
if (hasMathML || (isLaTeXSupported && /\$\$|\\\(/.test(fragmentHTML)))  {
    const mathJaxConfig = { renderLatex: isLaTeXSupported };
    await loadMathJax(mathJaxConfig);
    ...
}
```
That way mathJax gets loaded with the correct configuration we support regardless of whether it contains mathML or LaTeX.  It's entirely valid scenario of an HtmlBlock to initially contain mathML, and then the contents get updated to contain LaTeX after the first render.

Reply:
Ah yeah, I wasn't thinking of re-rendering. This makes sense, yeah.

Reply:
(i think this will address the other thread about the latex that is display: none above)

Reply:
It did and I was able to remove that. I got it stuck in my head that we only wanted the LaTeX component loaded if LaTeX was on the page.